target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-whw01-v1.dts

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "qcom-ipq4019-ap.dk01.1.dtsi"
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Linksys WHW01 v1";
+	compatible = "linksys,whw01-v1";
+
+	aliases {
+		led-boot = &led_system_blue;
+		led-running = &led_system_blue;
+	};
+
+	chosen {
+		bootargs-append = " root=/dev/ubiblock0_0";
+	};
+
+	reserved-memory {
```

Comment:
Are these needed?

Reply:
I'm unsure how to accurately confirm whether this is the case; I do not have access to QCOM datasheets for this board or SoC.

In the interest of caution these sections have been carved out based on the same `reserved-memory` sections present factory firmware device-tree. That said, these may have been for a Kernel built from a QCOM tree / BSP, and potentially for drivers not present in mainline? That said, it appears that the these additional sections were attempted to be upstreamed into `qcom-ipq4019-ap.dk01.1.dtsi` [previously](https://patchwork.kernel.org/project/linux-arm-kernel/patch/1521193101-4586-5-git-send-email-sricharan@codeaurora.org/).

An upper portion of `rsvd2` is already present in `qcom-ipq4019.dtsi` (`0x87e00000` to `0x88000000`), so this is only allocated as `0x300000` rather than `0x500000` - as the existing `smem` and `tz` entries in `qcom-ipq4019.dtsi` comprise the rest.

Reply:
You can simply remove them and try using the device, if it does not crash during that then these are not needed.
Usually, some of these are left for the crash log which the mainline does not use while others are already upstream.

Reply:
Removing these entries and rebuilding the image doesn't appear to have broken any functionality, so I guess they're not needed. I've pushed a new commit with them removed.